src/tape/ensemble.py

```diff
@@ -1199,6 +1203,108 @@ def from_parquet(
+        sync_tables=True,
+        additional_cols=True,
+        npartitions=None,
+        partition_size=None,
+        **kwargs,
+    ):
+        """Read in parquet file(s) for the object and source tables into an Ensemble object.
+
+        Parameters
+        ----------
+        source_file: 'str'
+            Path to a parquet file, or multiple parquet files that contain
+            source information to be read into the ensemble
+        object_file: 'str'
+            Path to a parquet file, or multiple parquet files that contain
+            object information.
+        column_mapper: 'ColumnMapper' object
+            If provided, the ColumnMapper is used to populate relevant column
+            information mapped from the input dataset.
+        provenance_label: 'str', optional
+            Determines the label to use if a provenance column is generated
+        sync_tables: 'bool', optional
+            In the case where object files are loaded in, determines whether an
+            initial sync is performed between the object and source tables. If
+            not performed, dynamic information like the number of observations
+            may be out of date until a sync is performed internally.
+        additional_cols: 'bool', optional
+            Boolean to indicate whether to carry in columns beyond the
+            critical columns, true will, while false will only load the columns
+            containing the critical quantities (id,time,flux,err,band)
+        npartitions: `int`, optional
+            If specified, attempts to repartition the ensemble to the specified
+            number of partitions
+        partition_size: `int`, optional
+            If specified, attempts to repartition the ensemble to partitions
+            of size `partition_size`.
+
+        Returns
+        ----------
+        ensemble: `tape.ensemble.Ensemble`
+            The ensemble object with parquet data loaded
+        """
+
+        # load column mappings
+        self._load_column_mapper(column_mapper, **kwargs)
+
+        # Handle additional columnss
+        if additional_cols:
+            columns = None  # None will prompt read_parquet to read in all cols
+        else:
+            columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
+            if self._provenance_col is not None:
+                columns.append(self._provenance_col)
+            if self._nobs_tot_col is not None:
+                columns.append(self._nobs_tot_col)
+            if self._nobs_band_cols is not None:
+                for col in self._nobs_band_cols:
+                    columns.append(col)
+
+        # Read in the source parquet file(s)
+        self.source = SourceFrame.from_parquet(source_file, index=self._id_col, columns=columns, 
+                                               ensemble=self)
+
+        # Read in the object file(s)
+        self.object = ObjectFrame.from_parquet(object_file, index=self._id_col, ensemble=self)
```

Comment:
It seems that we always do this for parquet files, but will take another look and think about it. Given the dataset size it seems better to not always do so.

Reply:
I'm fine with leaving as is and matching the behavior in from_parquet for now, but making a note of this since there are a few recent issues involving what loader functions we should have and what behavior/workflows we might expect.